src/Core/Silk.NET.SilkTouch/BoolMarshaller.cs

```diff
@@ -0,0 +1,119 @@
+                        case UnmanagedType.VariantBool:
+                            ctx.LoadTypes[index] = ctx.Compilation.GetSpecialType(SpecialType.System_Int16);
+                            break;
+                        case UnmanagedType.U1:
+                        default:
+                            ctx.LoadTypes[index] = ctx.Compilation.GetSpecialType(SpecialType.System_Byte);;
+                            break;
+                    }
+
+                    ExpressionSyntax @true;
+                    ExpressionSyntax @false;
+
+                    if (ctx.ParameterMarshalOptions[index]?.MarshalAs == UnmanagedType.VariantBool)
+                    {
+                        @true = LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(-1)); 
+                        @false = LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(0));
+                    }
+                    else
+                    {
+                        @true = LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(1));
+                        @false = LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(0));
+                    }
+
+                    @true = CastExpression(IdentifierName(ctx.LoadTypes[index].ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat)), @true);
+                    @false = CastExpression(IdentifierName(ctx.LoadTypes[index].ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat)), @false);
+
+                    var name = $"bmp{ctx.Slot}{index}";
+                    ctx.DeclareVariable(ctx.LoadTypes[index], name);
+                    ctx.SetParameterToVariableAndAssign(index, name, ConditionalExpression(ctx.ParameterExpressions[index], @true, @false));
+                }
+            }
+
+
+            var resultLocalName = $"boolName{ctx.Slot}res";
+            var processReturnType = !ctx.ReturnsVoid && SymbolEqualityComparer.Default.Equals(ctx.ReturnLoadType, ctx.Compilation.GetSpecialType(SpecialType.System_Boolean));
+            if (processReturnType)
+            {
+                    ctx.ReturnLoadType = ctx.Compilation.GetSpecialType(SpecialType.System_Byte);
+
+                    ctx.DeclareVariable(ctx.Compilation.GetSpecialType(SpecialType.System_Boolean), resultLocalName);
+            }
+
+            next();
+
+            if (processReturnType)
+            {
+                ctx.CurrentStatements = ctx.CurrentStatements.Append
+                (
+                    ExpressionStatement
+                    (
+                        AssignmentExpression
+                        (
+                            SyntaxKind.SimpleAssignmentExpression, IdentifierName(resultLocalName),
+                            BinaryExpression
+                            (
+                                SyntaxKind.EqualsExpression, ctx.ResultExpression,
+                                LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(1))
```

Comment:
```suggestion
                                @true
```

Reply:
what do you mean? `@true` isn't defined in this scope

Reply:
Oh whoops. In any case, sometimes this'll be `-1`